### PR TITLE
[release-v1.12] Add CI config for Knative Client 1.12 release

### PR DESCRIPTION
**Pull request by dsimansk**
/cc @rhuss @Kaustubh-pande

### CODE REVIEW

File: ci-operator/config/openshift-knative/client/openshift-knative-client-release-v1.12.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/bin/kn
+  to: knative-client
+- dockerfile_path: openshift/ci-operator/knative-images/client/Dockerfile.cliartifacts
+  inputs:
+    bin:
+      paths:
```
Comment (Kaustubh-pande) [inline ci-operator/config/openshift-knative/client/openshift-knative-client-release-v1.12.yaml:29]: @dsimansk  I think we also need kn-darwin-arm64?

Comment (dsimansk) [reply ci-operator/config/openshift-knative/client/openshift-knative-client-release-v1.12.yaml:29]: Doesn't really matter here.

**Comment by dsimansk** [comment]
/pj-rehearse pull-ci-openshift-knative-client-release-v1.12-e2e-aws-ocp-415

**Pull request merged**